model/metrics/dotnet-metrics.yaml

```diff
@@ -0,0 +1,190 @@
+    unit: "By"
+    stability: experimental
+
+  - id: metric.dotnet.gc.memory.commited
+    type: metric
+    metric_name: dotnet.gc.memory.commited
+    brief: >
+      The amount of committed virtual memory for the managed GC heap, as
+      observed during the latest garbage collection.
+    note: >
+      This metric reports the same values as calling
+      [`GC.GetGCMemoryInfo().TotalCommittedBytes`](https://learn.microsoft.com/dotnet/api/system.gcmemoryinfo.totalcommittedbytes).
+      Committed virtual memory may be larger than the heap size because it
+      includes both memory for storing existing objects (the heap size) and some
+      extra memory that is ready to handle newly allocated objects in the
+      future.
+    instrument: updowncounter
+    unit: "By"
+    stability: experimental
+
+  - id: metric.dotnet.gc.heap.size
+    type: metric
+    metric_name: dotnet.gc.heap.size
+    brief: >
+      The heap size (including fragmentation), as observed during the latest
+      garbage collection.
+    note: "This metric reports the same values as calling [`GC.GetGCMemoryInfo().GenerationInfo.SizeAfterBytes`](https://learn.microsoft.com/dotnet/api/system.gcgenerationinfo.sizeafterbytes)."
+    instrument: updowncounter
+    unit: "By"
+    stability: experimental
+    attributes:
+      - ref: dotnet.gc.heap.generation
+        requirement_level: required
+
+  - id: metric.dotnet.gc.heap.fragmentation
+    type: metric
+    metric_name: dotnet.gc.heap.fragmentation
+    brief: >
+      The heap fragmentation, as observed during the latest garbage collection.
+    note: >
+      This metric reports the same values as calling
+      [`GC.GetGCMemoryInfo().GenerationInfo.FragmentationAfterBytes`](https://learn.microsoft.com/dotnet/api/system.gcgenerationinfo.fragmentationafterbytes).
+    instrument: updowncounter
+    unit: "By"
+    stability: experimental
+    attributes:
+      - ref: dotnet.gc.heap.generation
+        requirement_level: required
+
+  - id: metric.dotnet.gc.pause.time
+    type: metric
+    metric_name: dotnet.gc.pause.time
+    brief: "The total amount of time paused in GC since the process has started."
+    note: "This metric reports the same values as calling [`GC.GetTotalPauseDuration()`](https://learn.microsoft.com/dotnet/api/system.gc.gettotalpauseduration)."
+    instrument: counter
+    unit: "s"
+    stability: experimental
+
+  - id: metric.dotnet.jit.compiled_il.size
+    type: metric
+    metric_name: dotnet.jit.compiled_il.size
+    brief: "Count of bytes of intermediate language that have been compiled since the process has started."
+    note: "This metric reports the same values as calling [`JitInfo.GetCompiledILBytes()`](https://learn.microsoft.com/dotnet/api/system.runtime.jitinfo.getcompiledilbytes)."
+    instrument: counter
+    unit: "By"
+    stability: experimental
+
+  - id: metric.dotnet.jit.compiled_method.count
+    type: metric
+    metric_name: dotnet.jit.compiled_method.count
```

Comment:
it's a plain counter, so following the pattern should be `dotnet.jit.compiled_methods`